### PR TITLE
Formatting update

### PR DESCRIPTION
**Pull request by benjaminjt**
Adds a configuration file for rustfmt that sets the indentation to 2 spaces, because it's pretty 😂 

Feel free to object.

See `.rustfmt.toml` for the only real change, the rest is just the result of `cargo fmt`.

### CODE REVIEW

**Review by joshleeb — APPROVED** [review]
fine by me :+1:

File: .rustfmt.toml [hunk 1]
```diff
@@ -0,0 +1 @@
+tab_spaces = 2
```
Comment (jordwest) [inline .rustfmt.toml]: Does not obey the `.editorconfig` change: `insert_final_newline = true`  😂

**Review by tobyjsullivan — COMMENTED** [review]
A bit late but LGTM 👍

**Comment by jordwest** [comment]
Looks like it's just @AKST

**Comment by benjaminjt** [comment]
Hey @tobyjsullivan 👋hope you're well!

**Pull request merged**